translatepy/translators/yandex.py

```diff
@@ -119,62 +73,49 @@ def translate(self, text, destination_language, source_language="auto") -> Union
                     return None, None
             if isinstance(source_language, Language):
                 source_language = source_language.yandex_translate
-            # check if we have an _sid
-            if self._sid.replace(" ", "") == "" and not self.refreshSID():
-                return None, None
 
             def _request():
-                """ """
-                url = self._base_url + "translate?id=" + self._sid + "-0-0&srv=tr-text&lang=" + str(source_language) + "-" + str(destination_language)  + "&reason=auto&format=text"
-                request = get(url, headers=self._headers, data={'text': str(text), 'options': '4'})
-                data = loads(request.text)
+                url = self._base_url + "translate?ucid=" + self._ucid() + "&srv=android" + "&format=text"
+                request = post(url, headers=HEADERS, data={'text': str(text), 'lang': str(source_language) + "-" + str(destination_language)})
+                data = request.json()
                 if request.status_code < 400 and data["code"] == 200:
-                    data = loads(request.text)
                     return str(data["lang"]).split("-")[0], data["text"][0]
-                return None, None
+                return None, None  # TODO: Raise exception by YT returned status code
 
             _lang, _text = _request()
-            if _lang is None and _text is None:
-                if self.refreshSID():
-                    _lang, _text = _request()
+
             return _lang, _text
         except Exception:
             return None, None
 
-    def transliterate(self, text, source_language=None) -> Union[Tuple[str, str], Tuple[None, None]]:
+    def transliterate(self, text, destination_language, source_language="auto") -> Union[Tuple[str, str], Tuple[None, None]]:
```

Comment:
I don't think that destination_language is really needed when you make a transliteration. We can just use a dummy language like "en".

Also, we should URL Encode the text like that to fix issues like #6 
```python
text = quote(str(text), safe='')
```

Reply:
Thanks, fixed